IO/src/main/java/io/deephaven/io/log/LogEntry.java

```diff
@@ -20,9 +20,9 @@ public interface LogEntry extends LogOutput, LogSink.Element {
 
     LogEntry start(LogSink sink, LogLevel level, long currentTimeMicros, Throwable t);
 
-    LogEntry end();
+    void end();
```

Comment:
This is a very clever solution. You aren't adding anything to *prevent* double frees, if users really want, however. I think you should at least add JavaDoc to the effect that `end()` or `endl()` must be invoked exactly once.